### ISSUE TITLE
Mongoose library deprecated. Can't get Bluebird library to work.

### ISSUE THREAD

**Issue by JohnsonLamb** [linnovate/mean#1681]
Hello.

## Information

- OS
 Windows 10 Home 64 bit

- nodejs
 4.6.0

- npm
 2.15.9

- mean status
_module.js:327
    throw err;
    ^
Error: Cannot find module 'npm'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\Users\Johns\AppData\Roaming\npm\node_modules\mean-cli\lib\cli.js:5:7)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)

## Problem:_

I have been setting up the MEAN developing environment following the instructions provided at mean.io and I have been having problems with **mongoose**.

The first problem occurred when I got to the point where you run the **Gulp** command in your App directory, I got two errors:

_Error: Cannot find module 'mongoose'_

_[nodemon] app crashed - waiting for file changes starting..._

When I go to the _http://localhost:3000_ page i get the follwing message:

_The localhost page isn’t working
localhost didn’t send any data.
ERR_EMPTY_RESPONSE_

I solved these two problems by installing mongoose module in my App directory by running the **npm install mongoose** command. However, I still get the same error when navigating to the localhost and at the console the following message is displayed:

_Mongoose: mpromise (mongoose's default promise library) is deprecated plug your own promise library instead: http://mongoosejs.com/docs/promises.html
GET / - - - - ms_

I tried the Cleaning the Cache (npm and bower) and Updated npm, gulp and bower. **Did not work**.

I did some more researching and decided to install the Bluebird promise library in my App folder (npm install bluebird). **Did not work**

On Bluebirds website it says to write the following line of code, in order to install the library:

_var Promise = require("bluebird");_

I don't know enough about nodejs to understand where this line goes but I still tried looking for .js files with the **"promise = require"** text. Found 2 files inside the Gulp folder, located in the root folder of the App: e2e.js & test.js. Inside both of them i found the follwing line of code:

**var promise = require('../server.js')**

I replaced it with the one from Bluebird's website. **It still did not work**

I don't know where to go from here, where do I insert this line?

Thank you for your time.

**Comment by timelf123** [comment]
To start, you will need npm > 3. Try wiping your node_modules folder after getting npm 3 and reinstalling

**Comment by JohnsonLamb** [comment]
Thank you for your reply. I was able to make the App work! Here's what I did:

I updated the npm to the new version using the **$ npm install npm@latest -g** command. I now have npm v3.10.4 running.

I then followed the steps described in Mean.io to create a new App:

Used mean-cli to create my App. There were the following warnings:

- swig@1.4.2 is no longer maintained.

- minimatch@2.0.10 deprecated

- graceful-fs@1.2.3 deprecated

- fsevents@^1.0.0 Skipped because it was an optional dependency.

- npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})


Updated minimatch & graceful-fs by running the **$ npm install <module_name>@latest -g** command.

Installed the latest mongoose

Ran the **Gulp** command

Went to local host and there was the initial mean portal. Now the real challenge begins!

Thank you very much.
